### PR TITLE
add stack op

### PR DESCRIPTION
**Pull request by mbrookhart**
## Description ##
Missing Op from GNMT

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by cconvey** [comment]
@mbrookhart Do we have a test case that exercises this PR?

**Comment by mbrookhart** [comment]
@cconvey tests/python/unittest/test_operator.py::test_stack hits this op with a few dozen examples.

**Comment by mbrookhart** [comment]
1. Yes, MXNet verifies before we get to this point.
2. That's a good point, updated

File: src/ngraph/ngraph_emitter.cc [hunk 1]
```diff
@@ -686,6 +686,25 @@ void Emitter::CreateLayerOps() {
...
 
+  // stack takes a list of tensors of equal shape and
+  // concatenates them along a given axis expanded for each input
+  ngraph_op_funcs_["stack"] = [this](const NodePtr& node) {
+    // get the concat axis
+    size_t axis = get_default_transformed_axis(node, "axis", 0,
+                                               node->inputs_[0]->shape_.ndim() + 1);
+    auto shape = op_map_[node->inputs_[0]]->get_shape();
+    shape.insert(shape.begin() + axis, 1);
+    // grab in input ngraph nodes and Reshape them
```
Comment (cconvey) [inline src/ngraph/ngraph_emitter.cc]: NBD, but IMO the code below is quite readable and so this comment (line 697) isn't necessary.

Also, is "in input" a typo?

Comment (mbrookhart) [reply src/ngraph/ngraph_emitter.cc]: Yep
